src/Model/Export/Catalog/ProductField/GenericField.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Omikron\Factfinder\Model\Export\Catalog\ProductField;
+
+use Magento\Catalog\Model\Product;
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Magento\Framework\Phrase;
+use Magento\Framework\ObjectManagerInterface;
+use Omikron\Factfinder\Api\Export\Catalog\ProductFieldInterface;
+use Omikron\Factfinder\Exception\AttributeExportException;
+use Psr\Log\LoggerInterface;
+
+class GenericField implements ProductFieldInterface
+{
+    /** @var string */
+    private $attributeName;
+
+    /** @var ScopeConfigInterface */
+    private $scopeConfig;
+
+    /** @var LoggerInterface */
+    private $logger;
+
+    public function __construct(
+        string $attributeName,
+        ScopeConfigInterface $scopeConfig,
+        ObjectManagerInterface $objectManager
+    ) {
+        $this->attributeName = $attributeName;
+        $this->scopeConfig   = $scopeConfig;
+        $this->logger        = $objectManager->get('Omikron\Factfinder\Export\Logger');
+    }
+
+    /**
+     * @param Product $product
+     *
+     * @return string
+     * @throws AttributeExportException
+     */
+    public function getValue(Product $product): string
+    {
+        try {
+            return (string) $product->getAttributeText($this->attributeName);
+        } catch (\Throwable $e) {
+            if ($this->scopeConfig->isSetFlag('factfinder/general/logging_enabled')) {
```

Comment:
I'd rather just throw the exception and catch it for example in the export script, i dont think logging belongs in this class

Reply:
I used the attribute repository. It throws NoSuchEntityException if attribute doesn't exist. With that, i think the Custom exception class i added is redundant.